diff --git a/src/kem/ntruprime/pqclean_sntrup761_clean/crypto_core_inv3sntrup761.c b/src/kem/ntruprime/pqclean_sntrup761_clean/crypto_core_inv3sntrup761.c
index 87f3750..7a6b5d8 100644
--- a/src/kem/ntruprime/pqclean_sntrup761_clean/crypto_core_inv3sntrup761.c
+++ b/src/kem/ntruprime/pqclean_sntrup761_clean/crypto_core_inv3sntrup761.c
@@ -39,13 +39,16 @@ static small F3_freeze(int16 x) {
     return (small) (x - 3 * ((10923 * x + 16384) >> 15));
 }
 
+#define p OQS_KEM_NTRUPRIME_p
 /* byte p of output is 0 if recip succeeded; else -1 */
 int PQCLEAN_SNTRUP761_CLEAN_crypto_core_inv3sntrup761(unsigned char *outbytes, const unsigned char *inbytes) {
     small *out = (void *) outbytes;
-    small *in = (void *) inbytes;
+    const small *in = (void *) inbytes;
     small f[p + 1], g[p + 1], v[p + 1], r[p + 1];
-    int i, loop, delta;
-    int sign, swap, t;
+    int i, loop;
+    int delta = 1;
+    int swap, t;
+    int16 sign;
 
     for (i = 0; i < p + 1; ++i) {
         v[i] = 0;
