diff --git a/src/kem/ntruprime/pqclean_sntrup761_clean/crypto_core_weightsntrup761.c b/src/kem/ntruprime/pqclean_sntrup761_clean/crypto_core_weightsntrup761.c
index 3809abf..950c21d 100644
--- a/src/kem/ntruprime/pqclean_sntrup761_clean/crypto_core_weightsntrup761.c
+++ b/src/kem/ntruprime/pqclean_sntrup761_clean/crypto_core_weightsntrup761.c
@@ -13,7 +13,7 @@ int PQCLEAN_SNTRUP761_CLEAN_crypto_core_weightsntrup761(unsigned char *outbytes,
     int16 weight = 0;
     int i;
 
-    for (i = 0; i < p; ++i) {
+    for (i = 0; i < OQS_KEM_NTRUPRIME_p; ++i) {
         weight += in[i] & 1;
     }
     PQCLEAN_SNTRUP761_CLEAN_crypto_encode_int16(outbytes, &weight);
